presto-main/src/main/java/io/prestosql/operator/PagesHash.java

```diff
@@ -168,6 +183,39 @@ public int getAddressIndex(int rightPosition, Page hashChannelsPage, long rawHas
         return -1;
     }
 
+    public void getAddressIndex(int firstRightPosition, int rightPositions, Page hashChannelsPage, long[] rawHashes, long[] result)
```

Comment:
1. Does this method proves to be more efficient than calling `getAddressIndex` in sequence?